pkg/controllers/localbuild/controller.go

```diff
@@ -581,6 +638,165 @@ func (r *LocalbuildReconciler) requestArgoCDAppSetRefresh(ctx context.Context) e
+	req, err := http.NewRequest("POST", argocdEndpoint+"/session", bytes.NewBuffer(payloadBytes))
+	if err != nil {
+		return fmt.Errorf("Error creating HTTP request: %v\n", err), "failed"
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	// Create an HTTP client and disable TLS verification
+	client := &http.Client{}
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.TLSClientConfig.InsecureSkipVerify = true
+	client.Transport = transport
+
+	// Send the request
+	resp, err := client.Do(req)
+	if err != nil {
+		return fmt.Errorf("Error sending request: %v\n", err), "failed"
+	}
+	defer resp.Body.Close()
+
+	// Read the response body
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("Error reading response body: %v\n", err), "failed"
+	}
+
+	if resp.StatusCode == 200 {
+		var argocdSession ArgocdSession
+
+		err := json.Unmarshal([]byte(body), &argocdSession)
+		if err != nil {
+			fmt.Errorf("Error unmarshalling JSON: %v", err)
+		}
+
+		payload := map[string]string{
+			"name":            "admin",
+			"currentPassword": adminPassword,
+			"newPassword":     argocdDevModePassword,
+		}
+
+		payloadBytes, err := json.Marshal(payload)
+		if err != nil {
+			return fmt.Errorf("Error creating JSON payload: %v\n", err), "failed"
+		}
+
+		req, err := http.NewRequest("PUT", argocdEndpoint+"/account/password", bytes.NewBuffer(payloadBytes))
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", argocdSession.Token))
+		req.Header.Set("Content-Type", "application/json")
+
+		resp, err := client.Do(req)
+		if err != nil {
+			return fmt.Errorf("Error sending request: %v\n", err), "failed"
+		}
+		defer resp.Body.Close()
+
+		// Lets checking the new admin password
+		payload = map[string]string{
+			"username": "admin",
+			"password": argocdDevModePassword,
+		}
+		payloadBytes, err = json.Marshal(payload)
+		if err != nil {
+			return fmt.Errorf("Error creating JSON payload: %v\n", err), "failed"
+		}
+
+		req, err = http.NewRequest("POST", argocdEndpoint+"/session", bytes.NewBuffer(payloadBytes))
+		if err != nil {
+			return fmt.Errorf("Error creating HTTP request: %v\n", err), "failed"
+		}
+		req.Header.Set("Content-Type", "application/json")
+
+		// Create an HTTP client and disable TLS verification
+		client := &http.Client{}
+		transport := http.DefaultTransport.(*http.Transport).Clone()
+		transport.TLSClientConfig.InsecureSkipVerify = true
+		client.Transport = transport
+
+		// Send the request
+		resp, err = client.Do(req)
+		if err != nil {
+			return fmt.Errorf("Error sending request: %v\n", err), "failed"
+		}
+		defer resp.Body.Close()
+
+		if resp.StatusCode == 200 {
+			// Password verification succeeded !
+			return nil, "succeeded"
+		} else {
+			return fmt.Errorf("### New password verification failed: %s", body), "failed"
+		}
+
+	} else {
+		return fmt.Errorf("HTTP Error: %d", resp.StatusCode), "failed"
+	}
+	return nil, "failed"
```

Comment:
What is the meaning of returning nil (indicating no error) but failed?

Reply:
Fixed: 700a841